### ISSUE TITLE
Allow pcluster ssh to use MasterPrivateIP

### ISSUE THREAD

**Issue by PhilipMorrisJones** [aws/aws-parallelcluster#1717]
**Environment:**
 - aws-parallelcluster-2.6.0
 - Configuration file: any

**Bug description and how to reproduce:**

The command `pcluster ssh` always uses MasterPublicIP. We use cloud9 to create clusters as we cannot install awscli locally and also have no access to the secret keys to configure the command line if we did.

We are not allowed to have port 22 open to the world so have a security group to address that. By adding the security group from the cloud9 instance it means that whatever IP address the cloud9 instance gets, it can connect to the master node of the cluster. **However**, ssh only works if I ssh to MasterPrivateIP and not MasterPublicIP and `pcluster ssh` always uses MasterPublicIP. Can there be a flag added to `pcluster ssh` to allow ssh to use MasterPrivateIP?

```
SupportOrg:~/.parallelcluster $ pcluster status pmjmon30-02
Status: CREATE_COMPLETE
MasterServer: RUNNING
MasterPublicIP: 34.236.85.109
ClusterUser: ec2-user
MasterPrivateIP: 10.0.0.93
SupportOrg:~/.parallelcluster $ pcluster ssh pmjmon30-02 -o ConnectTimeout=5
ssh: connect to host 34.236.85.109 port 22: Connection timed out
SupportOrg:~/.parallelcluster $ ssh ec2-user@34.236.85.109 -o ConnectTimeout=5
ssh: connect to host 34.236.85.109 port 22: Connection timed out
SupportOrg:~/.parallelcluster $ ssh ec2-user@10.0.0.93 -o ConnectTimeout=5                                                                                                                            
Last login: Mon Mar 30 15:36:40 2020 from ip-10-0-0-166.ec2.internal

       __|  __|_  )
       _|  (     /   Amazon Linux 2 AMI
      ___|\___|___|

https://aws.amazon.com/amazon-linux-2/
1 package(s) needed for security, out of 32 available
Run "sudo yum update" to apply all updates.
```

**Comment by lukeseawalker** [comment]
Hi @PhilipMorrisJones
I'm marking this as a feature request.

**Comment by lmouhib** [comment]
This would also be good to have for NICE DCV.

**Comment by enrico-usai** [comment]
This has been already addressed as part of [2.4.0 release](https://github.com/aws/aws-parallelcluster/blob/develop/CHANGELOG.md#240). The same is true for `pcluster dcv` command.
> pcluster ssh command now works for clusters with use_public_ips = false
